### ISSUE TITLE
Discrepancy between Comment and Code for Feature Concatenation in _cluster_features

### ISSUE THREAD

**Issue by skyswordw** [huangwl18/ReKep#38]
File: keypoint_proposal.py

Location: Inside the KeypointProposer class, within the _cluster_features method.
Description:

I've noticed a discrepancy between a comment and the subsequent code block regarding the features being added to the clustering input X.

The relevant code section is:
```
features_pca = torch.mm(obj_features_flat, v[:, :3])
features_pca = (features_pca - features_pca.min(0)[0]) / (features_pca.max(0)[0] - features_pca.min(0)[0])
X = features_pca
# add feature_pixels as extra dimensions  <--- COMMENT IN QUESTION
feature_points_torch = torch.tensor(feature_points, dtype=features_pca.dtype, device=features_pca.device)
feature_points_torch  = (feature_points_torch - feature_points_torch.min(0)[0]) / (feature_points_torch.max(0)[0] - feature_points_torch.min(0)[0])
X = torch.cat([X, feature_points_torch], dim=-1)
# cluster features to get meaningful regions
cluster_ids_x, cluster_centers = kmeans(
    X=X,
    # ...
)
```

The comment "# add feature_pixels as extra dimensions" suggests that 2D pixel coordinates (feature_pixels, which are defined earlier in the method as feature_pixels = np.argwhere(binary_mask)) are intended to be concatenated with the DINOv2 visual features (features_pca) for clustering.

However, the actual code concatenates feature_points_torch to X. The variable feature_points_torch is derived from feature_points, which are the 3D world coordinates of the points within the current mask (feature_points = points[binary_mask]).

Potential Impact:

This isn't necessarily a bug in functionality, as concatenating 3D world coordinates with visual features for clustering is a valid approach and can be beneficial for spatial discrimination. However, the comment is misleading and doesn't reflect the current implementation.
